core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -824,22 +822,16 @@ class KafkaController(val config: KafkaConfig, zkUtils: ZkUtils, val brokerState
   }
 
   private def updateLeaderEpochAndSendRequest(topicAndPartition: TopicAndPartition, replicasToReceiveRequest: Seq[Int], newAssignedReplicas: Seq[Int]) {
-    brokerRequestBatch.newBatch()
     updateLeaderEpoch(topicAndPartition.topic, topicAndPartition.partition) match {
       case Some(updatedLeaderIsrAndControllerEpoch) =>
         try {
+          brokerRequestBatch.newBatch()
```

Comment:
It seems like this should be in the try/catch since it can throw an IllegalStateException and it seems like `updateLeaderEpoch` doesn't do anything with this field. But please double-check.